packages/unchained-client/src/ethereum/parser/uniV2.ts

```diff
@@ -131,6 +142,38 @@ export class Parser implements SubParser {
     }
   }
 
+  async parseStakingRewards(tx: BlockbookTx): Promise<TxSpecific | undefined> {
+    const txData = tx.ethereumSpecific?.data
+
+    if (!txData) return
+
+    const txSigHash = getSigHash(txData)
+
+    if (!Object.values(this.supportedStakingRewardsFunctions).some((hash) => hash === txSigHash))
+      return
+
+    const decoded = this.stakingRewardsInterface.parseTransaction({ data: txData })
+
+    // failed to decode input data
+    if (!decoded) return
+
+    return {
+      data: {
+        parser: TxParser.UniV2,
+        method: decoded.name
+      }
+    }
+  }
+
+  async parse(tx: BlockbookTx): Promise<TxSpecific | undefined> {
+    if (txInteractsWithContract(tx, UNI_V2_ROUTER_CONTRACT)) {
+      return this.parseUniV2(tx)
+    } else if (txInteractsWithContract(tx, UNI_V2_STAKING_REWARDS)) {
+      return this.parseStakingRewards(tx)
+    }
+    return
+  }
```

Comment:
We can tighten this up a bit:

```suggestion
  async parse(tx: BlockbookTx): Promise<TxSpecific | undefined> {
    if (txInteractsWithContract(tx, UNI_V2_ROUTER_CONTRACT)) return this.parseUniV2(tx)
    if (txInteractsWithContract(tx, UNI_V2_STAKING_REWARDS)) return this.parseStakingRewards(tx)
    return
  }
```